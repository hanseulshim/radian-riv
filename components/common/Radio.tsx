import React from 'react'

interface Props {
  value: string
  label: string
  checked: boolean
  name?: string
  onChange: (e: any) => void
  className?: string
  disabled?: boolean
}

export default function Radio({
  value,
  label,
  checked,
  name,
  onChange,
  className,
  disabled
}: Props) {
  return (
    <div className={`radio ${className || ''}`}>
      <label className={disabled ? 'disabled' : ''}>
        <input
          type="radio"
          value={value}
          checked={checked}
          onChange={onChange}
          disabled={disabled}
          name={name}
        />
        {label}
      </label>
    </div>
  )
}
