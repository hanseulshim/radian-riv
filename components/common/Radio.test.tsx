import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Radio from './Radio'

describe('Radio', () => {
  let component
  beforeAll(() => {
    const props = {
      value: 'test',
      label: 'test',
      onChange: () => {},
      checked: false
    }
    component = shallow(<Radio {...props} />)
  })
  it('renders radio', () => {
    expect(component.find('.radio')).toHaveLength(1)
  })

  it('renders input', () => {
    expect(component.find('.radio > label > input')).toHaveLength(1)
  })
  it('renders label', () => {
    expect(component.find('label').text()).toEqual('test')
  })

  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
