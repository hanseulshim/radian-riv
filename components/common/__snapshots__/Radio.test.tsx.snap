// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should match snapshot 1`] = `
<div
  className="radio "
>
  <label
    className=""
  >
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="test"
    />
    test
  </label>
</div>
`;
